kion/resource_project_enforcement.go

```diff
@@ -0,0 +1,326 @@
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Threshold value. Either a dollar amount or a percentage, depending on the threshold type.",
+			},
+			"cloud_rule_id": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Defines a Cloud Rule ID associated with the enforcement.",
+			},
+			"notification_frequency": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Frequency at which notifications are sent for this enforcement.",
+			},
+			"project_id": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "ID of the project under enforcement.",
+			},
+			"ou_id": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "If enforcement is from an Organizational Unit (OU), this is the ID of the OU.",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Flag that specifies if the enforcement is enabled.",
+			},
+			"overburn": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Flag that specifies if enforcement will place project in an overburn state when triggered.",
+			},
+			"user_group_ids": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+				Optional:    true,
+				Description: "List of user group IDs that will receive notifications from the enforcement.",
+			},
+			"user_ids": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+				Optional:    true,
+				Description: "List of user IDs that will receive notifications from the enforcement.",
+			},
+			"triggered": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Flag that specifies if the enforcement is currently triggered.",
+			},
+		},
+	}
+}
+
+func resourceProjectEnforcementCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	client := m.(*hc.Client)
+
+	projectID, ok := d.GetOk("project_id")
+	if !ok {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Invalid or Missing Project ID",
+			Detail:   "The project ID is either missing or not valid. Please provide a valid project ID.",
+		})
+		return diags
+	}
+	projectIDInt := projectID.(int)
+
+	post := hc.ProjectEnforcementCreate{
+		Description:   d.Get("description").(string),
+		Timeframe:     d.Get("timeframe").(string),
+		SpendOption:   d.Get("spend_option").(string),
+		AmountType:    d.Get("amount_type").(string),
+		ServiceID:     hc.OptionalInt(d, "service_id"),
+		ThresholdType: d.Get("threshold_type").(string),
+		Threshold:     d.Get("threshold").(int),
+		CloudRuleID:   hc.OptionalInt(d, "cloud_rule_id"),
+		Overburn:      d.Get("overburn").(bool),
+		UserGroupIds:  convertToIntSlice(d.Get("user_group_ids").([]interface{})),
+		UserIds:       convertToIntSlice(d.Get("user_ids").([]interface{})),
+	}
```

Comment:
If you make more attributes optional per earlier comments, don't forget to use the optional helpers here too.